src/main/java/org/apache/brooklyn/cloudfoundry/entity/VanillaCloudfoundryApplicationImpl.java

```diff
@@ -381,16 +334,56 @@ public Boolean call() {
         }
     }
 
-    private String getLocalPath(String uri) {
-        try {
-            File war;
-            war = LocalResourcesDownloader
-                    .downloadResourceInLocalDir(uri);
-            return war.getCanonicalPath();
-        } catch (IOException e) {
-            log.error("Error obtaining local path in {} for artifact {}", this, uri);
-            throw Exceptions.propagate(e);
+    private void initDriver(CloudFoundryPaasLocation location) {
+        VanillaPaasApplicationDriver newDriver = doInitDriver(location);
+        if (newDriver == null) {
+            throw new UnsupportedOperationException("cannot start " + this +
+                    " on " + location + ": no driver available");
         }
+        driver = newDriver;
+    }
+
+    private VanillaPaasApplicationDriver doInitDriver(CloudFoundryPaasLocation location) {
+        if (driver != null) {
+            if ((driver instanceof VanillaPaasApplicationCloudFoundryDriver)
+                    && location.equals((driver).getLocation())) {
```

Comment:
Actually, A `VanillaCloudFundryApplicationDriver#getLocation()` returns a `CloudFoundryPaasLocation` (which is a `PaasLocation`). so, may it be redundant?